Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp

```diff
@@ -424,6 +445,9 @@ WebKit::WebsiteDataStore& webkitWebsiteDataManagerGetDataStore(WebKitWebsiteData
             WebKit::WebsiteDataStore::defaultHSTSDirectory() : FileSystem::stringFromFileSystemRepresentation(priv->hstsCacheDirectory.get()));
         configuration->setMediaKeysStorageDirectory(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory());
         configuration->setLocalStorageQuota(priv->localStorageQuota);
+        if (priv->perOriginStorageQuota > 0)
+            configuration->setPerOriginStorageQuota(priv->perOriginStorageQuota);
+        // else use default from StorageQuotamanager (that is 1G)
```

Comment:
Ah, so the default is 1G. Where is that coming from?

Reply:
Yes, the default is 1000MB that comes from StorageQuotaManager (https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/wpe-2.28/Source/WebCore/storage/StorageQuotaManager.h):
`    static constexpr uint64_t defaultThirdPartyQuotaFromPerOriginQuota(uint64_t quota) { return quota / 10; }
    static constexpr uint64_t defaultQuota() { return 1000 * MB; }
    static constexpr uint64_t defaultThirdPartyQuota() { return defaultThirdPartyQuotaFromPerOriginQuota(defaultQuota()); }
`
This is why I decided to use 0 as special case to preserve current quota default is new api is not used.

And this 5MB fix is here because StorageQuotaManager will ask UIClient for quota increase if it run out of storage space and enlarge the quota to whatever UIClient provides (max(uiClientQuota,currentQuota)). This means that setting anything below 5MB as per-origin-storage-quota will be overrided eventually with UIClient with it's 5MB but I doubt we would ever want to use such small values.